### PR TITLE
implemented functional increment/decrement buttons on pill button

### PR DESCRIPTION
**Pull request by JoHeyJo**
incrementing/decrementing updates the database and updates the UI to reflect the quantity amount in the database.

### CODE REVIEW

**Comment by lananelson** [comment]
wow this is pretty convenient! I think this design will work well! will need to do a bit of polish, e.g. `- and +` look a bit small and off-center, and having the full circle around them is a bit off maybe only outer border is needed

functionality-wise the number in the pill should match the actual quantity and then you can remove the quantity that is under the product name, of course then we will have to add quantity unit somewhere as well which can be a bit weird to have inside of the pill

**Comment by lananelson** [comment]
Looking very sleek! There is a small edge cases to fix:
- negative increment of quantity is possible

File: components/PillButton.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from 'react';
+
+function PillButton({ id, quantity, unit, updateQuantity }) {
```
Comment (lananelson) [inline components/PillButton.tsx:3]: thats a nice and simple component!

File: pages/pantries/[id].tsx [hunk 2]
```diff
@@ -4,6 +4,11 @@ import { supabase } from '../../api';
 import ProductEditor from '../../components/ProductEditor';
 import SlideOver from '../../components/SlideOverDialog';
 import Filter from '../../components/Filter';
+import PillButton from '../../components/PillButton'
+
+function cl(...a) {
+  console.log(a);
+}
```
Comment (lananelson) [inline pages/pantries/[id].tsx:11]: don't forget to remove! we really need a eslint rule and check for this

File: pages/pantries/[id].tsx [hunk 3]
```diff
@@ -36,6 +41,7 @@ export default function Pantry() {
       .from('pantries')
       .select(`*, products(*)`)
       .filter('id', 'eq', id)
+      .order(`id`, { foreignTable: 'products' })
```
Comment (lananelson) [inline pages/pantries/[id].tsx:44]: nice

File: pages/pantries/[id].tsx [hunk 4]
```diff
@@ -121,6 +127,19 @@ export default function Pantry() {
     }
   }
 
+  async function updateQuantity(id, currentProductQuantity) {
+    const { data, error } = await supabase
+      .from('products')
+      .update({ quantity_amount: currentProductQuantity })
+      .eq('id', id)
+    if (error) {
+      setErrorMessages(errorMessages => [...errorMessages, error]);
```
Comment (lananelson) [inline pages/pantries/[id].tsx:136]: how can I view an error message?

Comment (JoHeyJo) [reply pages/pantries/[id].tsx:136]: @lananelson not sure what you mean.

Comment (lananelson) [reply pages/pantries/[id].tsx:132]: if this error happens where in UI will I see it? how can I reproduce to test this out?

**Comment by lananelson** [comment]
@JoHeyJo instead of disabling minus button when the quantity is 0 it might be better to hide it; disabled look is not super obvious

**Comment by lananelson** [comment]
I think decrementing to 0 is essentially the same as marking something out of stock so it seems to me that the functionality of the pill button is pretty much complete

**Pull request merged**